fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/impl/attachment/AttachmentPersistentState.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.attachment;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.world.PersistentState;
+
+/**
+ * Backing storage for server-side world attachments.
+ * Thanks to custom [.isDirty] logic, the file is only written if something needs to be persisted.
+ */
+public class AttachmentPersistentState extends PersistentState {
+	public static final String ID = "fabric_data_attachments";
```

Comment:
Question: Should this `fabric:data_attachments` ?

Should this include some sort of version to allow for possible changes later down the line? Maybe a version should go into the NBT? Or not at all?

Reply:
unsure of the general conventions about this tbh, are there any other modules that add sections to NBT like this?

Reply:
That's the convention for JSON, so I guess I'll change it accordingly